pulumi-eks-go/main.go

```diff
@@ -0,0 +1,439 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/pulumi/pulumi-aws/sdk/v4/go/aws/ec2"
+	"github.com/pulumi/pulumi-aws/sdk/v4/go/aws/eks"
+	"github.com/pulumi/pulumi-aws/sdk/v4/go/aws/iam"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		prefix := "pulumi-eks-go"
+
+		// Resource: VPC
+		// Purpose: Amazon Virtual Private Cloud (Amazon VPC) enables you to launch AWS resources into a virtual network that you've defined.
+		// Docs: https://docs.aws.amazon.com/vpc/latest/userguide/what-is-amazon-vpc.html
+
+		// VPC CIDR
+		cidrBlock := "10.0.0.0/16"
+
+		// VPC Args
+		vpcArgs := &ec2.VpcArgs{
+			CidrBlock:          pulumi.String(cidrBlock),
+			EnableDnsHostnames: pulumi.Bool(true),
+			InstanceTenancy:    pulumi.String("default"),
+			Tags: pulumi.StringMap{
+				"Name": pulumi.String(prefix + "-vpc"),
+			},
+		}
+
+		// VPC
+		vpc, err := ec2.NewVpc(ctx, prefix+"-vpc", vpcArgs)
+		if err != nil {
+			fmt.Println(err.Error())
+			return err
+		}
+
+		// Resource: Subnets
+		// Purpose: A subnet is a range of IP addresses in your VPC.
+		// Docs: https://docs.aws.amazon.com/vpc/latest/userguide/configure-subnets.html
+
+		// 3 Private Subnets
+		privSubnet1, err := ec2.NewSubnet(ctx, prefix+"-priv-subnet-1", &ec2.SubnetArgs{
+			VpcId:            vpc.ID(),
+			CidrBlock:        pulumi.String("10.0.1.0/24"),
+			AvailabilityZone: pulumi.String("us-east-2a"),
```

Comment:
Future ToDo: Dynamically get subnets after creating VPC - see `ec2.GetSubnetIds` - explicit is fine for this example at the moment